### DISCUSSION TITLE
Using Custom Extensions

### DISCUSSION THREAD

**Discussion by Degorap** [donmccurdy/glTF-Transform#573] [Q&A] [answered]
Hi, I'm trying to use the CLI to merge multiple glTFs, but they have a custom 3rd party extension.
What do I have to do to make the CLI use a custom extension? or is it not possible?
And if it is not possible, how else could I merge glTFs with this type of extensions?
sorry for the newbie question. 
D.

**Comment by donmccurdy** [comment, marked as answer]
For glTF-Transform to preserve a third-party extension, an implementation of that extension using the [extension API](https://gltf-transform.donmccurdy.com/extensions.html#custom-extensions) must be registered. Currently the CLI doesn't support registering custom extensions but I hope to allow that, see #85. Until that is implemented, you'd need to register the extension and execute the merge using the scripting API instead:

```typescript
import { Document, NodeIO } from '@gltf-transform/core';
import { KHRONOS_EXTENSIONS } from '@gltf-transform/extensions';
import { MyCustomExtension } from './path/to/extension';

// Register extensions.
const io = new NodeIO()
  .registerExtensions([...KHRONOS_EXTENSIONS, MyCustomExtension]);

// Merge documents.
const document = new Document();
for (const path of paths) {
  document.merge(await io.read(path));
}

// (Optional) Merge buffers.
const buffer = document.getRoot().listBuffers()[0];
document.getRoot()
  .listAccessors()
  .forEach((a) => a.setBuffer(buffer));
document.getRoot()
  .listBuffers()
  .forEach((b, index) => (index > 0 ? b.dispose() : null));

// Write.
await io.write('output.gltf', document);
```

**Reply by Degorap** [reply]
Looks complex for my skills, but anyway I will try, Thanks you!!
